woocommerce/woocommerce-android
# [Order Attribution] Add attribution to orders created from the mobile app

Closes: #10990 

Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2964

### Description
This PR adds a source_type attribution to the orders created using the mobile app, this will allow both the app and the web to correctly show the origin of the order as "Mobile App".

@ThomazFB @atorresveiga I'm pinging you  to double check the commit 911cfcf227bdc6ea7ae43dbf6e7928e4cb32f9cc, I noticed the two functions `createOrUpdateDraft` and `placeOrder` had almost the same logic, and from what I remember `createOrUpdateDraft` was added just to make sure we use the status `auto-draft` for draft orders, while now the status is synced too (change done here https://github.com/woocommerce/woocommerce-android/pull/6709), so there is no need to have two separate functions that do the same thing, and which could be confusing for people non-familiar with the code, WDYT? I think this is a safe change, but if I'm missing some cases, please let me know and I can revert it.

### Testing instructions
1. Optional for the step 4: make sure your store is running the latest WC beta 8.7.0-beta.2 (using https://woo.com/products/woocommerce-beta-tester/)
2. Create an order using the app.
3. Check the order in the app and confirm the correct origin is shown.
4. Optional: Open the order in wp-admin and confirm the correct origin is shown.

### Images/gif
| App | wp-admin |
| --- | --- |
| <img width=400, src="https://github.com/woocommerce/woocommerce-android/assets/1657201/3bb89bc8-8470-4c40-8a1e-e9e7824be781"/> | <img width="687" alt="Screenshot 2024-03-05 at 11 15 28" src="https://github.com/woocommerce/woocommerce-android/assets/1657201/23bf2f50-1842-4c50-9871-53c98118872c"> |
